### PR TITLE
Fabo/857 addressbook

### LINKED ISSUE #857: Create a peer store
UI Version: 0.7.0

Description:

Relying on peers in the config.toml is error prone. Especially on testnets. I think we should store discovered peers and cycle through them when trying to connect.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #857

Description:

introduce an addressbook that stores found peers so we do not use the hard coded ips everytime and therefor also avoid crashing voyager if those go down.

- [x] write tests

❤️ Thank you!

### CODE REVIEW

File: app/src/main/addressbook.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    subPeersHostnames
+      // check if we already know the peer
+      .filter(subPeerHostname => !this.peerIsKnown(subPeerHostname))
+      // add new peers to state
+      .forEach(subPeerHostname => {
+        this.addPeer(subPeerHostname)
+      })
+
+    this.persistToDisc()
```
Comment (okwme) [inline app/src/main/addressbook.js]: don't need to `persistToDisc()` if `!subPeersHostnames.length`

File: app/src/main/addressbook.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    this.peers = []
+
+    this.loadFromDisc()
+
+    // add persistent peers to already stored peers
+    persistent_peers
+      .filter(peer => !this.peerIsKnown(peer))
+      .forEach(peer => this.addPeer(peer))
+
+    this.persistToDisc()
```
Comment (okwme) [inline app/src/main/addressbook.js]: don't need to `persistToDisc()` if `!persistent_peers.length`

File: test/unit/specs/main.spec.js [hunk 3]
```diff
@@ -173,26 +173,32 @@ describe("Startup Process", () => {
...
+          }
+      )
+
       Object.assign(process.env, {
         COSMOS_NODE: "123.456.789.123"
       })
       // run main
-      main = await require(appRoot + "src/main/index.js")
-
-      expect(
```
Comment (okwme) [inline test/unit/specs/main.spec.js]: why would these not pass anymore?

File: test/unit/specs/main.spec.js [hunk 4]
```diff
@@ -173,26 +173,32 @@ describe("Startup Process", () => {
...
-        childProcess.spawn.mock.calls.find(
-          ([path, args]) =>
-            path.includes("gaiacli") &&
-            args.includes("rest-server") &&
-            args.includes("123.456.789.123:46657")
-        )
-      ).toBeDefined()
```
Comment (okwme) [inline test/unit/specs/main.spec.js]: why would these not pass anymore?

**Comment by faboweb** [comment]
All very good comments @okwme. Made me rethink some things. :)

File: test/unit/specs/main.spec.js (hunk 4 above)
Comment (faboweb) [reply test/unit/specs/main.spec.js]: I put this logic now into the addressbook

**Pull request merged**